test/language/block-scope/syntax/leave/verify-context-in-finally-block-1.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) Copyright 2011 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 13.1
+description: >
+    verify context in finally block 1
+---*/
+function f() {}
+
+try {
+  (function(x) {
+    try {
+      let x = 'inner';
+      throw 0;
+    } finally {
+      f();
+      assert.sameValue(x, 'outer');
+    }
+  })('outer');
+} catch (e) {}
```

Comment:
Maybe to avoid the extra logic of conditionally re-throwing, it would be better in these cases to simply capture the value inside the `try` block and make the assertion afterward:


```diff
+var result;
 try {
  (function(x) {
    try {
      let x = 'inner';
      throw 0;
    } finally {
      f();
-     assert.sameValue(x, 'outer');
+     result = x;
     }
   })('outer');
 } catch (e) {}
+assert.sameValue(result, 'outer');
```

...but then again, since this test concerns variable scoping directly, it might be dangerous to use the closure like this.